src/Psy/TabCompletion/Matchers/AbstractMatcher.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Psy\TabCompletion\Matchers;
+
+use Psy\Context;
+use Psy\ContextAware;
+
+/**
+ * Class AbstractMatcher
+ * @package Psy\TabCompletion\Matchers
+ */
+abstract class AbstractMatcher implements ContextAware
+{
+    /** Syntax types */
+    const CONSTANT_SYNTAX = '^[A-Z0-9-_]+$';
+    const VAR_SYNTAX = '^\$[a-zA-Z0-9-_]+$';
```

Comment:
How about using the [real regex](http://php.net/manual/en/language.variables.basics.php), `[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*` ?